### PR TITLE
[submodules] Update grpc to v1.40.0

### PR DESCRIPTION
**Pull request by townsend2010**

### CODE REVIEW

File: 3rd-party/submodule_info.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ info on where to find release information about the submodule.
 **This file needs updating when a submodule is updated!**
 
 ### gRPC
-Version: 1.36.4 (+[our patches](https://github.com/CanonicalLtd/grpc/compare/v1.36.4..cb8af401)) |
+Version: 1.40.0 (+[our patches](https://github.com/CanonicalLtd/grpc/compare/v1.40.0..310f4367)) |
```
Comment (ricab) [inline 3rd-party/submodule_info.md:10]: Same here, we don't seem to have v1.40 in our repo.

**Review by ricab — COMMENTED** [review]
Link not working on this one either.

File: 3rd-party/submodule_info.md (hunk 1 above)
Comment (townsend2010) [reply 3rd-party/submodule_info.md:10]: D'oh!  Fixed, thanks!

**Review by ricab — COMMENTED** [review]
Looks good, thanks for the update.

bors r+

**Review by ricab — APPROVED** [review]

**Comment by townsend2010** [comment]
bors retry

**Pull request merged**